Compilation error for FATFS

### LVGL version

v9.1.0

### What happened?

Enabled LV_USE_FS_FATFS in lv_conf,h and got the following error:
.pio/libdeps/esp32-s3-devkitc-1/lvgl/src/libs/fsdrv/lv_fs_fatfs.c:228:5: error: unknown type name 'DIR'; did you mean 'DDR'?


### How to reproduce?

#define LV_USE_FS_FATFS 1
#if LV_USE_FS_FATFS
    #define LV_FS_FATFS_LETTER 'F'     /*Set an upper cased letter on which the drive will accessible (e.g. 'A')*/
    #define LV_FS_FATFS_CACHE_SIZE 0    /*>0 to cache this number of bytes in lv_fs_read()*/
#endif

--- a/src/libs/fsdrv/lv_fs_fatfs.c
+++ b/src/libs/fsdrv/lv_fs_fatfs.c
@@ -17,7 +17,7 @@
  *********************/
 
 #ifdef ESP_PLATFORM
-#define DIR FF_DIR  /* ESP IDF typedefs `DIR` as `FF_DIR` in its version of ff.h. Use `FF_DIR` in LVGL too */
+    #define DIR FF_DIR  /* ESP IDF typedefs `DIR` as `FF_DIR` in its version of ff.h. Use `FF_DIR` in LVGL too */
 #endif
 
 #if LV_FS_FATFS_LETTER == '\0'

--- a/src/libs/fsdrv/lv_fs_fatfs.c
+++ b/src/libs/fsdrv/lv_fs_fatfs.c
@@ -9,6 +9,9 @@
 #include "../../../lvgl.h"
 
 #if LV_USE_FS_FATFS
+
+#define FF_DIR DIR    /* ESP IDF typedefs `DIR` as `FF_DIR`. Rename to `DIR` for use with this source file */
+#define FATFS_DIR DIR /* Mbed OS typedefs `DIR` as `FATFS_DIR`. Rename to `DIR` for use with this source file */
 #include "ff.h"
 
 #include "../../core/lv_global.h"

--- a/src/libs/fsdrv/lv_fs_fatfs.c
+++ b/src/libs/fsdrv/lv_fs_fatfs.c
@@ -9,15 +9,16 @@
 #include "../../../lvgl.h"
 
 #if LV_USE_FS_FATFS
-
-#define FF_DIR DIR    /* ESP IDF typedefs `DIR` as `FF_DIR`. Rename to `DIR` for use with this source file */
-#define FATFS_DIR DIR /* Mbed OS typedefs `DIR` as `FATFS_DIR`. Rename to `DIR` for use with this source file */
 #include "ff.h"
 
 #include "../../core/lv_global.h"
 /*********************
  *      DEFINES
  *********************/
+
+#ifdef ESP_PLATFORM
+#define DIR FF_DIR  /* ESP IDF typedefs `DIR` as `FF_DIR` in its version of ff.h. Use `FF_DIR` in LVGL too */
+#endif
 
 #if LV_FS_FATFS_LETTER == '\0'
     #error "LV_FS_FATFS_LETTER must be an upper case ASCII letter"

